### DISCUSSION TITLE
hola programadores

### DISCUSSION THREAD

**Discussion by maximo6984** [community/community#122287] [Programming Help]
### Body

hola programadores queria preguntar si alguien podria echarme una mano con ese codico que lo voy a poner mas abajo..el codico es para una estrategia de trading para la plataforma ctrader no lo e creado yo si no el chatGPT yo solo lo copie pero en la plataforma de bots cuando le doy compillar  me sale unos errores y no se como podria solucionarla... voy a poner todo el codico y los errore ..alomejor me podria ayudar alquien ..

using cAlgo.API;
using cAlgo.API.Internals;
using cAlgo.API.Indicators;
using System;

namespace cAlgo.Robots
{
    [Robot(TimeZone = TimeZones.UTC, AccessRights = AccessRights.None)]
    public class HullMovingAverageCross : Robot
    {
        private HullMovingAverage hullMA20;
        private HullMovingAverage hullMA200;

        protected override void OnStart()
        {
            hullMA20 = Indicators.HullMovingAverage(MarketSeries.Close, 20);
            hullMA200 = Indicators.HullMovingAverage(MarketSeries.Close, 200);
        }

        protected override void OnTick()
        {
            if (IsLastBarOpen)      ---  ERROR ME PONE QUE NO EXIXSTE 
                return;

            if (hullMA20.Result.Last(1) > hullMA200.Result.Last(1) && hullMA20.Result.Last(2) <= hullMA200.Result.Last(2))
            {
                ExecuteMarketOrder(TradeType.Buy, Symbol, VolumeInUnits, "Buy Order", 0, 0, null, "HullMA20Cross");
            }
            else if (hullMA20.Result.Last(1) < hullMA200.Result.Last(1) && hullMA20.Result.Last(2) >= hullMA200.Result.Last(2))
            {
                ExecuteMarketOrder(TradeType.Sell, Symbol, VolumeInUnits, "Sell Order", 0, 0, null, "HullMA20Cross");
            }
        }
    }
}

Y TAMBIEN ME PONE ERROR  CON EL VolumenIUnits que no existe...
gracias de antemano

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by maximo6984** [comment]
using cAlgo.API;
using cAlgo.API.Internals;
using cAlgo.API.Indicators;
using System;

namespace cAlgo.Robots
{
    [Robot(TimeZone = TimeZones.UTC, AccessRights = AccessRights.None)]
    public class HullMovingAverageCross : Robot
    {
        private HullMovingAverage hullMA20;
        private HullMovingAverage hullMA200;

        protected override void OnStart()
        {
            hullMA20 = Indicators.HullMovingAverage(MarketSeries.Close, 20);
            hullMA200 = Indicators.HullMovingAverage(MarketSeries.Close, 200);
        }

        protected override void OnTick()
        {
            if (hullMA20.Result.LastValue > hullMA200.Result.LastValue && hullMA20.Result.IsRising())
            {
                ExecuteMarketOrder(TradeType.Buy, Symbol, GetVolumeInUnits(), "Buy Order", 0, 0, null, "HullMA20Cross");
            }
            else if (hullMA20.Result.LastValue < hullMA200.Result.LastValue && hullMA20.Result.IsFalling())
            {
                ExecuteMarketOrder(TradeType.Sell, Symbol, GetVolumeInUnits(), "Sell Order", 0, 0, null, "HullMA20Cross");
            }
        }

        private double GetVolumeInUnits()
        {
            // Aquí puedes definir tu propia lógica para determinar el volumen de la orden
            return 10000; // Por ejemplo, 10000 unidades
        }
    }
}
este es otro codico pero tampoco funcciona

**Comment by Avotexs** [comment]
proba este:         
using cAlgo.API;
using cAlgo.API.Internals;
using cAlgo.API.Indicators;
using System;

namespace cAlgo.Robots
{
    [Robot(TimeZone = TimeZones.UTC, AccessRights = AccessRights.None)]
    public class HullMovingAverageCross : Robot
    {
        private HullMovingAverage hullMA20;
        private HullMovingAverage hullMA200;
        private double takeProfitPips;
        private double stopLossPips;

        protected override void OnStart()
        {
            hullMA20 = Indicators.HullMovingAverage(MarketSeries.Close, 20);
            hullMA200 = Indicators.HullMovingAverage(MarketSeries.Close, 200);

            // Set your desired TakeProfit and StopLoss values (in pips)
            takeProfitPips = 100.0;
            stopLossPips = 50.0;
        }

        protected override void OnTick()
        {
            if (hullMA20.Result.LastValue > hullMA200.Result.LastValue && hullMA20.Result.IsRising())
            {
                ExecuteMarketOrder(TradeType.Buy, Symbol, Symbol.VolumeInLots, "Buy Order", takeProfitPips, stopLossPips, null, "HullMA20Cross");
            }
            else if (hullMA20.Result.LastValue < hullMA200.Result.LastValue && hullMA20.Result.IsFalling())
            {
                ExecuteMarketOrder(TradeType.Sell, Symbol, Symbol.VolumeInLots, "Sell Order", takeProfitPips, stopLossPips, null, "HullMA20Cross");
            }
        }
    }
}

**Comment by ebndev** [comment]
Hi @maximo6984, thanks for participating!

It looks like you were able to get some answers to your question, hopefully it helps!

Please note: We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.
